### PR TITLE
fix: [multi-slb] Put the service in the load balancer that has no lab…

### LINKED ISSUE #5261: Multiple Standard LoadBalancers doesn't respect ServiceLabelSelector
#### What happened:

Having the below SLB config to distribute Service creation between 3 Azure Load Balancers:
```json
"multipleStandardLoadBalancerConfigurations": [
  {
    "name": "kubernetes",
    "primaryVMSet": "sys"
  },
  {
    "name": "nginx1",
    "primaryVMSet": "nginx1"
    "ServiceLabelSelector": {
      "matchLabels": {
        "app.kubernetes.io/name": "nginx"
      }
    }
  },
  {
    "name": "nginx2",
    "primaryVMSet": "nginx2"
    "ServiceLabelSelector": {
      "matchLabels": {
        "app.kubernetes.io/name": "nginx"
      }
    }
  },
  {
    "name": "nginx3",
    "primaryVMSet": "nginx3"
    "ServiceLabelSelector": {
      "matchLabels": {
        "app.kubernetes.io/name": "nginx"
      }
    }
  }
]
```

And creating a `Service` with label `app.kubernetes.io/name: ingress-nginx` makes all 4 SLB configs eligable for placement.

#### What you expected to happen:

Only the SLB configs matching the `ServiceLabelSelector` should be eligable.

#### How to reproduce it (as minimally and precisely as possible):

1. Run `cloud-controller-manager` with `--v=10` or higher (or attach to debugger)
2 .Appy the following manifests

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/name: nginx
  name: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app.kubernetes.io/name: nginx
  template:
    metadata:
      labels:
        app.kubernetes.io/name: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.25
        env:
        - name: NGINX_PORT
          value: "8080"
        ports:
        - containerPort: 8080
          name: http
          protocol: TCP
      nodeSelector:
        kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: nginx1
  annotations:
    service.beta.kubernetes.io/azure-load-balancer-internal: "false"
  labels:
    app.kubernetes.io/name: nginx
spec:
  type: LoadBalancer
  selector:
    app.kubernetes.io/name: nginx
  ports:
  - port: 80
    targetPort: http
---
apiVersion: v1
kind: Service
metadata:
  name: nginx2
  annotations:
    service.beta.kubernetes.io/azure-load-balancer-internal: "false"
  labels:
    app.kubernetes.io/name: nginx
spec:
  type: LoadBalancer
  selector:
    app.kubernetes.io/name: nginx
  ports:
  - port: 80
    targetPort: http
---
apiVersion: v1
kind: Service
metadata:
  name: nginx3
  annotations:
    service.beta.kubernetes.io/azure-load-balancer-internal: "false"
  labels:
    app.kubernetes.io/name: nginx
spec:
  type: LoadBalancer
  selector:
    app.kubernetes.io/name: nginx
  ports:
  - port: 80
    targetPort: http
```

3. Notice that the `kubernetes` SLB is eligable for placement

#### Anything else we need to know?:

Issue tested on version 1.28.4.

The issue lays in how eligability based on selectors are implemented. Currently they only remove SLB's with selectors from the list when that's the case. And in the case of having a service with a label or namespace selector it always returns the complete list of SLB's.

The expected behaviour is that if there's a label match, only the matches SLB's are eligable.

#### Environment:

- Kubernetes version (use `kubectl version`): 1.26.11
- Cloud provider or hardware configuration: Self-managed Kubernetes on Azure with CAPZ
- OS (e.g: `cat /etc/os-release`): Ubuntu 22.04.3 LTS
- Kernel (e.g. `uname -a`): 6.2.0-1017-azure 17~22.04.1-Ubuntu SMP 
- Install tools: CAPZ
- Network plugin and version (if this is a network-related bug): N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by nilo19**
…el/namespace selector only if there is no other choice for the service.

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

For a load balancer with no label/namespace selectors:
current behavior: it can be a placement targets for services.
new behavior: it can be a placement target for a service if and only if there is no other choice for the service.
Ref: https://github.com/kubernetes-sigs/cloud-provider-azure/issues/5261

#### Which issue(s) this PR fixes:

Fixes #5261

#### Does this PR introduce a user-facing change?

```release-note
fix: [multi-slb] Put the service in the load balancer that has no label/namespace selector only if there is no other choice for the service.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
/lgtm

thanks for the fixes!

**Comment by feiskyer** [comment]
please validate manually as well and unhold after that.

/hold

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -3810,7 +3810,7 @@ func getMostEligibleLBForService(
 	ruleCount := 301
 	if existingLBs != nil {
 		for _, existingLB := range *existingLBs {
-			if StringInSlice(pointer.StringDeref(existingLB.Name, ""), eligibleLBs) {
+			if StringInSlice(trimSuffixIgnoreCase(pointer.StringDeref(existingLB.Name, ""), consts.InternalLoadBalancerNameSuffix), eligibleLBs) {
```
Comment (nilo19) [inline pkg/provider/azure_loadbalancer.go:3813]: We need to ignore the -internal suffix when comparing lb names, or we will ignore all of the internal lbs here.

**Comment by nilo19** [comment]
@desek this would fix the issue. Please take a look.

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**